build/task-build-deps.js

```diff
@@ -43,19 +78,33 @@ async function findNativeModules() {
 async function rebuild() {
   console.log('Rebuilding modules...');
   const command = path.join(__dirname, '..', 'node_modules', '.bin', 'electron-rebuild');
-  await BuildUtils.spawn(command, ['-f'], {
+  await BuildUtils.spawn(command, [
+    '-f',
+    '-e', BuildUtils.getElectronRoot(),
+    '-v', BuildUtils.getElectronVersion(),
+  ], {
     stdio: 'inherit',
   });
 }
 
 export default async function() {
   let existingConfig = {};
   try {
-    existingConfig = await fs.readJson(BUILD_CONFIG_JSON);
+    existingConfig = await fs.readJson(BuildUtils.getBuildConfigFile());
   } catch (e) {
     // Missing files mean we rebuild
   }
 
+  const electron = BuildUtils.getManifest()._electron;
+  try {
+    if (electron.version !== BuildUtils.getElectronVersion()) {
```

Comment:
Sorry, I'm still not clear.  I mean, don't catch and re-download on error.  If there's a bad ZIP, don't redownload it!  If the "error case" includes bad versions (which I don't think it does, since you're testing the version above) that's very misleading and should be a separate `catch` case at least.